exhibitor-core/src/main/java/com/netflix/exhibitor/core/config/azure/AzureConfigProvider.java

```diff
@@ -0,0 +1,147 @@
+    }
+
+    public AzureClient getAzureClient() {
+        return azureClient;
+    }
+
+    @Override
+    public void start() throws Exception {
+        // NOP
+    }
+
+    @Override
+    public void close() throws IOException {
+        azureClient.close();
+    }
+
+    @Override
+    public PseudoLock newPseudoLock() throws Exception {
+        return new AzurePseudoLock
+                (
+                        azureClient,
+                        arguments.getContainer(),
+                        arguments.getLockArguments().getPrefix(),
+                        arguments.getLockArguments().getTimeoutMs(),
+                        arguments.getLockArguments().getPollingMs(),
+                        arguments.getLockArguments().getSettlingMs()
+                );
+    }
+
+    @Override
+    public LoadedInstanceConfig loadConfig() throws Exception {
+        Date lastModified;
+        Properties properties = new Properties();
+        CloudBlob object = getConfigObject();
+        if (object != null) {
+            lastModified = object.getProperties().getLastModified();
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            object.download(out);
+            properties.load(new ByteArrayInputStream(out.toByteArray()));
+        } else {
+            lastModified = new Date(0L);
+        }
+
+        PropertyBasedInstanceConfig config = new PropertyBasedInstanceConfig(properties, defaults);
+        return new LoadedInstanceConfig(config, lastModified.getTime());
+    }
+
+    @Override
+    public LoadedInstanceConfig storeConfig(ConfigCollection config, long compareVersion) throws Exception {
+        {
+            BlobProperties blobProperties = getConfigBlobProperties();
+            if (blobProperties != null) {
+                Date lastModified = blobProperties.getLastModified();
+                if (lastModified.getTime() != compareVersion) {
+                    return null;    // pattern copied from S3ConfigProvider - Azure may support a better way
+                }
+            }
+        }
+
+        PropertyBasedInstanceConfig propertyBasedInstanceConfig = new PropertyBasedInstanceConfig(config);
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        propertyBasedInstanceConfig.getProperties().store(out, "Auto-generated by Exhibitor " + hostname);
+
+        byte[] bytes = out.toByteArray();
+        azureClient.putBlob(bytes, arguments.getContainer(), arguments.getUri());
+
+        BlobProperties blobProperties = azureClient.getBlobProperties(arguments.getContainer(), arguments.getUri());
+        return new LoadedInstanceConfig(propertyBasedInstanceConfig, blobProperties.getLastModified().getTime());
+    }
+
+    private BlobProperties getConfigBlobProperties() throws Exception {
+        try {
+            BlobProperties properties = azureClient.getBlobProperties(arguments.getContainer(), arguments.getUri());
+            if (properties.getLength() > 0) {
```

Comment:
I don't like this pattern; why returning `null` when instead an empty `Properties` would be just as expressive?
if no properties is a serious exceptional situation that warrants people shutting and listen... then throw an exception :)